src/jit/codegenarm64.cpp

```diff
@@ -2470,7 +2470,8 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 // We unconditionally throw a divide by zero exception
                 genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
 
-                // We still need to call genProduceReg
+                // We still need to call genConsumeOperands/genProduceReg for liveness update.
+                genConsumeOperands(treeNode->AsOp());
```

Comment:
There is a but in else case too.  The model is that operands are consumed first and it will result in either reloading or copy into a reg if an operand is spilled or in a different register respectively.  Next operands are used to produce a result in target register.

In the else case (i.e. divisor not constant zero or gtOper = GT_UDIV), divisorReg and dividendReg are getting used even before genConsumeReg() is called on operands by genCodeForBinary().  This can lead to SBCG if any of the operands needs a reload/copy.  This would happen under register pressure.

I would suggest to move out genConsumeOperands() call from genCodeForBinary() and let the callers of genCodeForBinary() explicitly make the call to genConsumeOperands() at the right place.